### PR TITLE
feat: adds jobs to snmp discovery

### PR DESCRIPTION
**Pull request by jajeffries**
This pull request introduces a robust job tracking system for SNMP discovery policy executions, enabling better observability and status reporting of policy runs. The changes add a `JobStore` for managing job lifecycles, integrate job tracking into the policy execution flow, and expose policy/job statuses via the server API. Comprehensive unit tests are also included to ensure correctness and concurrency safety.

**Job Management and Tracking:**

- Added a new `JobStore` in `job.go` to manage job creation, status updates, and retrieval per policy, limiting history to the last 5 jobs per policy. The `Job` struct tracks status, errors, and timestamps.
- Integrated `JobStore` into the `Manager` and `Runner` components, ensuring each policy execution creates and updates a job record reflecting its outcome. [[1]](diffhunk://#diff-97d32253107e3c811caf8c6de6078b90fc18fbfecca66691424c657d8b77651cR35) [[2]](diffhunk://#diff-97d32253107e3c811caf8c6de6078b90fc18fbfecca66691424c657d8b77651cR53) [[3]](diffhunk://#diff-f0a20465993d1870b397eaebe9560ece5bad63ec7978b8e7c343a215ccf71ecdR46-R50) [[4]](diffhunk://#diff-f0a20465993d1870b397eaebe9560ece5bad63ec7978b8e7c343a215ccf71ecdR64) [[5]](diffhunk://#diff-f0a20465993d1870b397eaebe9560ece5bad63ec7978b8e7c343a215ccf71ecdR93-R98) [[6]](diffhunk://#diff-f0a20465993d1870b397eaebe9560ece5bad63ec7978b8e7c343a215ccf71ecdR131-R151)

**Policy Status Reporting:**

- Added a `PolicyStatus` struct and a `GetPolicyStatuses` method to `Manager`, aggregating the latest job status and history for each policy.
- Modified the server's `/status` endpoint to include policy/job statuses in the response, providing better visibility into policy execution state. [[1]](diffhunk://#diff-6b3406fdd3e26ab7dd44a8fae75ae1153bdfa88502cabb06e2481cf82ed6f5e3R32-R37) [[2]](diffhunk://#diff-6b3406fdd3e26ab7dd44a8fae75ae1153bdfa88502cabb06e2481cf82ed6f5e3L138-R150)

**Testing and Reliability:**

- Introduced extensive tests for `JobStore` covering creation, updating, concurrency, and edge cases, ensuring thread safety and correctness.
- Updated runner tests to validate job creation and status updates for successful, failed, and no-entity runs. [[1]](diffhunk://#diff-bbc860544a5e41d7a2f0276de913788d6bd8f73ea0f41553da62db94e8fe466fR100-R103) [[2]](diffhunk://#diff-bbc860544a5e41d7a2f0276de913788d6bd8f73ea0f41553da62db94e8fe466fR177) [[3]](diffhunk://#diff-bbc860544a5e41d7a2f0276de913788d6bd8f73ea0f41553da62db94e8fe466fL195-R197) [[4]](diffhunk://#diff-bbc860544a5e41d7a2f0276de913788d6bd8f73ea0f41553da62db94e8fe466fL220-R231) [[5]](diffhunk://#diff-bbc860544a5e41d7a2f0276de913788d6bd8f73ea0f41553da62db94e8fe466fR245) [[6]](diffhunk://#diff-bbc860544a5e41d7a2f0276de913788d6bd8f73ea0f41553da62db94e8fe466fL286-R294) [[7]](diffhunk://#diff-bbc860544a5e41d7a2f0276de913788d6bd8f73ea0f41553da62db94e8fe466fR335-R347) [[8]](diffhunk://#diff-bbc860544a5e41d7a2f0276de913788d6bd8f73ea0f41553da62db94e8fe466fL365-R379) [[9]](diffhunk://#diff-bbc860544a5e41d7a2f0276de913788d6bd8f73ea0f41553da62db94e8fe466fR402-R406)

These changes collectively improve the observability, reliability, and testability of policy executions in the SNMP discovery service.

### CODE REVIEW

File: snmp-discovery/policy/runner.go [hunk 1]
```diff
@@ -120,20 +127,27 @@ func (r *Runner) run() {
 
 	if len(entities) == 0 {
 		r.logger.Info("No entities to ingest", slog.Any("policy", r.ctx.Value(policyKey)))
+		// Update job status to completed even if no entities
+		r.jobStore.UpdateJob(policyName, job.ID, JobStatusCompleted, nil)
```
Comment (leoparente) [inline snmp-discovery/policy/runner.go]: @jajeffries should we report the number of ingested entities in the job completed? maybe that could be useful info

Comment (jajeffries) [reply snmp-discovery/policy/runner.go]: I wondered about that and actually put it in originally, but took it out again. It's not trivial to work out how many entities are in the request because some are nested and there is potential for duplicates.

Comment (leoparente) [reply snmp-discovery/policy/runner.go]: I agree, but at least the top level ones will give is the sense if something was discovered or not

**Review by leoparente — APPROVED** [review]
